src/main/java/jenkins/branch/NoTriggerBranchProperty.java

```diff
@@ -83,7 +108,12 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
                                     } else {
                                         for (BranchProperty prop : ((MultiBranchProject) j.getParent()).getProjectFactory().getBranch(j).getProperties()) {
                                             if (prop instanceof NoTriggerBranchProperty) {
-                                                return false;
+                                                NoTriggerBranchProperty triggerProperty = (NoTriggerBranchProperty) prop;
+                                                boolean suppressIndexing = SuppressionStrategy.EVENTS != triggerProperty.getStrategy();
+                                                boolean suppressEvents = SuppressionStrategy.INDEXING != triggerProperty.getStrategy();
```

Comment:
NIT: at first glance this logic is a little hard to parse and looks like it can be optimized but it can't

```suggestion
                                                boolean suppressIndexing = SuppressionStrategy.EVENTS != triggerProperty.getStrategy(); // INDEXING or ALL
                                                boolean suppressEvents = SuppressionStrategy.INDEXING != triggerProperty.getStrategy(); // EVENTS or ALL
```

or add an abstract  `BranchCause` between `Cause` and the `Branch*Cause`s (or a marker interface) and move the logic to the enum with a `shouldSupress(BranchCause)`

Reply:
I would like to not add comments at the end of the line because code should be self-descriptive 🙂I added the `shouldSuppress(Cause)` method. I haven't added the marker interface yet. Instead I added a logic responsible for checking if a correct type was passed. Based on the logic it has no chance to be executed, however it is a public API, so it probably has sense. Let me know what do you think.

If you prefer, I can add the marker interface but then we cannot prevent situations when somebody add it to completely other type (the suppression logic could start behaving strange).